sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentProducer.java

```diff
@@ -81,7 +81,11 @@ void populatePartitionedQueryMetrics() {
                                 fetchExecutionRangeAccumulator.getExecutionRanges(),
                                 Arrays.asList(schedulingTimeSpanMap)
                         ), pageResult.getActivityId());
-                BridgeInternal.putQueryMetricsIntoMap(pageResult, feedRange.getRange().toString(), qm);
+                String partitionKeyRange = pageResult.getResponseHeaders().get(HttpConstants.HttpHeaders.PARTITION_KEY_RANGE_ID);
+                if(StringUtils.isEmpty(partitionKeyRange)) {
+                    partitionKeyRange = "0";
```

Comment:
@mbhaskar please have a look and see if this proper way of fetching partition id